### PR TITLE
Disable HttpTelemetry for HTTP 3 requests

### PR DESCRIPTION
**Pull request by MihaZupan**
Avoid logging orphaned RequestStart events when using HTTP3, as the Http3 logic is not instrumented with Telemetry.

Contributes to #37428

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs [hunk 1]
```diff
@@ -359,7 +359,7 @@ public ValueTask<HttpResponseMessage> SendProxyConnectAsync(HttpRequestMessage r
 
         public ValueTask<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool async, bool doRequestAuth, CancellationToken cancellationToken)
         {
-            return HttpTelemetry.Log.IsEnabled() && request.RequestUri != null ?
+            return HttpTelemetry.Log.IsEnabled() && request.Version.Major < 3 && request.RequestUri != null ?
```
Comment (ManickaP) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs:362]: Should there be a tracking issue and a link saying we need to remove this condition once we add telemetry for H3? Or what is the plan for H3?

Comment (MihaZupan) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs:362]: I opened a tracking issue for this: #40896

Comment (ManickaP) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs:362]: Could you add comment here with the link, something like:
```
// [ActiveIssue("https://github.com/dotnet/runtime/issues/40896")]
```

**Review by ManickaP — APPROVED** [review]
LGTM, thanks.

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs (hunk 1 above)
Comment (MihaZupan) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs]: Added

**Pull request merged**